Xenomorph Tweaks

GunHog opened the pull request:
All PR contents on behalf of @WJohn.

:cl: WJohnston, Gun Hog
tweak: Xenos are now considered female.
tweak: Hunter pounce are now always blocked by shields, and may be blocked more often by other protective items.
tweak: Alien Queen is now somewhat faster.
tweak: Adjusted larva growth sprites.
tweak: Removed 20% facehugger failure chance for masks.
tweak: Disarm now always forces a person to drop an item in the active hand, and if a person's hand is empty/undroppable, Disarm will always result in a tackle.
tweak: Cyborgs disarmed by a Xeno will first disable their selected module, and if there is none selected, the cyborg will be pushed back and stunned.
tweak: Tail Sweep now has a slightly longer stun.
/:cl:

Commit:
Xenomorph Tweaks

- Xenos are now considered female.
- (Nerf) Hunter pounce are now always blocked by shields, and may be
blocked more often by other protective items.
- (Buff) Alien Queen is 2 points faster.
- Adjusted larva growth sprites.
- Removed 20% facehugger failure chance for masks.
- (Tweak) Disarm now always forces a person to drop an item in the
active hand, and if a person's hand is empty/undroppable, Disarm will
always result in a tackle.
- (Nerf) Borgs are no longer stunned by Xeno disarms.
- (Buff) Tail Sweep now has a slightly longer stun.

phil235 commented on code/modules/mob/living/carbon/alien/humanoid/caste/hunter.dm:
```diff
@@ -68,13 +68,15 @@
 				sleep(2)//Runtime prevention (infinite bump() calls on hulks)
 				step_towards(src,L)
+			else
+				weakened = 2
```
I'm pretty sure this is not how you're supposed to modify the weakened var.

GunHog replied:
I will fix it in the next commit.

GunHog replied:
It turns out that this is correct. Weaken(2) only slows Xenos, it does not stun them.

X-TheDark replied:
This is because aliens don't have CANWEAKEN flag in their status_flags.
Mobs have it by default : https://github.com/tgstation/tgstation/blob/master/code/modules/mob/mob_defines.dm#L124
But aliens override that : https://github.com/tgstation/tgstation/blob/master/code/modules/mob/living/carbon/alien/alien.dm#L27

optimumtact replied:
Just waiting on you to do this for this to be merged.

phil235 commented on code/modules/mob/living/carbon/alien/alien.dm:
```diff
@@ -7,7 +7,7 @@
 	voice_name = "alien"
 	icon = 'icons/mob/alien.dmi'
-	gender = NEUTER
+	gender = FEMALE //All xenos are girls!!
```
So examining a xeno will show "she" instead of "it"??

GunHog replied:
Well, that is the intent, however it does not, because examine code is trash. Take a look at how humans and carbons are done.

WJohn commented:
Thank youuuu :+1:

ShadowDeath6 commented:
So how are xenos supposed to deal with borgs? Do they just swipe at them until they die or do they have another way of stunning?

WJohn commented:
Swipe at them until they die. You deal 20 brute a swing, after all.

This also finally makes borgs a useful viable counter to xenos without getting completely shut down by stunlocking, as they can act as a get-in get-out role, dragging and releasing plasma or freeing and saving crew while taking a beating.

Alternatively it could perhaps be changed where disarm behaves much the same way it does with them as it does humans: Forces them to stow their active module first, THEN stunlocking begins.

PKPenguin321 commented:
In your cl,
>Xenos are now considered female.

Needs a prefix (tweak or add)

ShadowDeath6 commented:
I'm all fine and dandy for no stunlocking, but there's going to be a problem if we get a bunch of VTEC engiborg assholes running in with plasma, hitting themselves, then bailing out to heal themselves.

GunHog commented:
>In your cl,

Xenos are now considered female.
Needs a prefix (tweak or add)

Well spotted, fixed. 

@WJohn tell me if you want to change the borg disarm mechanics.

Davdichan commented:
Does pounce still turbo stun borgs for 5+ seconds?

AdamElTablawy commented:
>VTEC welderbombing engiborg squad

Eh xenos do need a counter I suppose but i'd prefer if tackle at the least unequipped the active module instead of it being a battle of dps between xeno and borg

Commit:
Teaks and Fixes

- Pouncing is now a LEAP_ATTACK, and is now ALWAYS blocked by shields.
- Changed Disarm combat for borgs: First disarm strike disables the
borg's active module. The second strike pushes the borg back and stuns
it.

GunHog commented:
New in this commit:
- Pouncing is now a LEAP_ATTACK, and is now ALWAYS blocked by shields.
- Changed Disarm combat for borgs: First disarm strike disables the
borg's active module. The second strike pushes the borg back and stuns
it.

Ultimate-Chimera commented:
A 6-8 foot tall beast should at least have a small chance to pounce through shields, they are huge, and probably heavy too.

PKPenguin321 commented:
Muh realism

WJohn commented:
Its not about realism, chimera, it's about consistency and gameplay. A riot shield should either block a xeno or not.

I actually wanted it so that riot shields would be guaranteed to block *all* thrown projectiles (including leaping xenos, which are considered thrown objects). A bullet is too fast, a melee attack can be swerved around your shield. A thrown object goes in a very obvious straight line/arc and isn't that fast, it should be a guaranteed block.

But that's for someone else to do.

Ultimate-Chimera commented:
Oh.

Davdichan commented:
Not to mention 'realistic' or 'lore friendly' xenos would die/collapse after being hit with a shotgun or high caliber weapon and melt a hole through the floor. Our Xenos are only based on the franchise.

GunHog commented:
>I actually wanted it so that riot shields would be guaranteed to block all thrown projectiles (including leaping xenos, which are considered thrown objects). A bullet is too fast, a melee attack can be swerved around your shield. A thrown object goes in a very obvious straight line/arc and isn't that fast, it should be a guaranteed block.

Xeno pounces are no longer considered thrown objects - this is to facilitate the balance change for shields to give 100% block to pounce. It is outside the scope of this PR to balance things unrelated to Xenos, so I made a new combat type for them. (LEAP_ATTACK)

>Not to mention 'realistic' or 'lore friendly' xenos would die/collapse after being hit with a shotgun or high caliber weapon and melt a hole through the floor. Our Xenos are only based on the franchise.

Hits (I do not know how many) from a combat shotgun, loaded with buckshot, at point blank will waste a Xeno. Xenos that die as a result of brute damage dropping acid on the hull to breach it sounds hilarious and something I would love to code. (Not in this PR though)

Commit:
Mini Fix

Commit:
Get that one too
